frontend/packages/git-service/src/utils/pac-strategy-detector.ts

```diff
@@ -0,0 +1,54 @@
+import { BaseService } from '../services/base-service';
+
+export const detectPacFiles = async (gitService: BaseService) => {
+  let pacFiles = [];
+  const isTektonFolderPresent = await gitService.isTektonFolderPresent();
+
+  if (isTektonFolderPresent) {
+    pacFiles.push('.tekton');
+    const { files: pipelineFiles } = await gitService.getRepoFileList({
+      specificPath: '.tekton',
+      includeFolder: true,
+    });
+    if (pipelineFiles.length > 0) {
+      pacFiles = [...pacFiles, ...pipelineFiles];
+    }
+  }
+  return pacFiles;
+};
+
+export const verifyPipelinesFile = async (file: string, gitService: BaseService) => {
+  const trimmedFile = file?.startsWith('.tekton/') ? file?.replace('.tekton/', '') : file;
+
+  const resourceContent = await gitService.getFileContent(`.tekton/${trimmedFile}`);
+  return resourceContent.includes('kind: PipelineRun');
+};
+
+export const isPipelineFilePresent = (pacFiles: string[], gitService: BaseService): boolean => {
+  if (!pacFiles || pacFiles?.length === 0) {
+    return false;
+  }
+  const pipelineFile: string = pacFiles.find(
+    (file) =>
+      (file?.endsWith('.yaml') || file?.endsWith('.yml')) && verifyPipelinesFile(file, gitService),
```

Comment:
`verifyPipelinesFile` returns a promise and the find here doesn't (and can not in a simple `array#find`) handle this, right? The promise is always truthy and doesn't check if the file contains "kind: Pipeline".

I think you should handle this in a for-loop with await and mark `isPipelineFilePresent` also as async. And this results in more async functions that call this, and so on.